modules/stack/main.tf

```diff
@@ -1,3 +1,26 @@
+locals {
+  triggers = [for trigger in var.triggers : "depends-on:${trigger}|state:FINISHED"]
+
+  imports = [for import in var.imports : "import:${import}"]
+
+  component_stack_deps = [for dep in var.component_stack_deps : "stack-deps:${dep}" if var.process_component_stack_deps == true]
+
+  stack_config_name_parts = split("-", var.stack_config_name)
+
+  folders = [
+    try(format("folder:%s", local.stack_config_name_parts[0]), ""),
+    try(format("folder:%s/%s", local.stack_config_name_parts[0], local.stack_config_name_parts[1]), ""),
```

Comment:
@aknysh this seems to require that stack files always have two parts. Is there any way you can make this more ambivalent to stack file naming conventions? Either by not using the file name at all and pulling the `environment` + `stage` from the config itself or at least by supporting as many `-` parts as the stack file includes?

Reply:
@Gowiem  thanks, I agree that we are hardcoding the "two-parts" here.
I'll look what could be done, but in this module, in the Rego policies, we had hardcoded the "two-parts" anyway, so fixing this here will not make the module more generic.
But good catch, thanks

Reply:
Longer term, we want to move to a Kubernetes Resource type of configuration where the component parts are described inside the file and the file names (and directory paths) can be arbitrary. In the short term, we have the file name convention`${environment}-${stage}` baked into a lot of places.